### PR TITLE
Add Document Level Security support for Opendistro role

### PR DESCRIPTION
**Pull request by JoupainMD**
In the same way as Field Level Security, I propose to add support for Document Level Security in Opendistro Role.
See documentation here https://opendistro.github.io/for-elasticsearch-docs/docs/security/access-control/document-level-security/

### CODE REVIEW

File: es/resource_elasticsearch_opendistro_role.go [hunk 1]
```diff
@@ -43,6 +43,14 @@ func resourceElasticsearchOpenDistroRole() *schema.Resource {
 							},
 							Set: schema.HashString,
 						},
+						"dls": {
```
Comment (phillbaker) [inline es/resource_elasticsearch_opendistro_role.go:46]: I didn't realize that we abbreviated `fls` before, it's a bit confusing, can we use `document_level_security` here?

**Review by phillbaker — APPROVED** [review]
looks good, two things:
* would like to fix the naming
* add an entry to the changelog

thanks!

**Comment by phillbaker** [comment]
It looks like this build is failing on travis in ES 7:
```
2020/11/06 15:51:37 [INFO] Failed to create OpenDistroRole: error creating role mapping: elastic: Error 500 (Internal Server Error): Cannot deserialize instance of `java.lang.String` out of START_ARRAY token

 at [Source: N/A; line: -1, column: -1] (through reference chain: com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleV7["index_permissions"]->java.util.ArrayList[0]->com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleV7$Index["dls"]) [type=mismatched_input_exception]: []

2020/11/06 15:51:37 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: error creating role mapping: elastic: Error 500 (Internal Server Error): Cannot deserialize instance of `java.lang.String` out of START_ARRAY token

 at [Source: N/A; line: -1, column: -1] (through reference chain: com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleV7["index_permissions"]->java.util.ArrayList[0]->com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleV7$Index["dls"]) [type=mismatched_input_exception]: []

2020/11/06 15:51:37 [ERROR] <root>: eval: *terraform.EvalSequence, err: error creating role mapping: elastic: Error 500 (Internal Server Error): Cannot deserialize instance of `java.lang.String` out of START_ARRAY token

 at [Source: N/A; line: -1, column: -1] (through reference chain: com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleV7["index_permissions"]->java.util.ArrayList[0]->com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleV7$Index["dls"]) [type=mismatched_input_exception]: []

    TestAccElasticsearchOpenDistroRole: testing.go:683: Step 0 error: errors during apply:

        

        Error: error creating role mapping: elastic: Error 500 (Internal Server Error): Cannot deserialize instance of `java.lang.String` out of START_ARRAY token

         at [Source: N/A; line: -1, column: -1] (through reference chain: com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleV7["index_permissions"]->java.util.ArrayList[0]->com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleV7$Index["dls"]) [type=mismatched_input_exception]: []
```

**Comment by JoupainMD** [comment]
Ok for changing the name, although it will take time for me to understand how to do it 🤕.  I also noticed that in the opendistro Documentation, dls is expected to be a string and not an array. see this [example](https://opendistro.github.io/for-elasticsearch-docs/docs/security/access-control/document-level-security/#parameter-substitution) on the documentation. Sorry it took time for me to test it, now it should work fine, I also added in readme an indication that one should use jsonencode to use document_level_security attributes.

File: es/util.go [hunk 2]
```diff
@@ -319,8 +319,11 @@ func flattenIndexPermissions(permissions []IndexPermissions) []map[string]interf
 		if len(permission.IndexPatterns) > 0 {
 			p["index_patterns"] = flattenStringSet(permission.IndexPatterns)
 		}
-		if len(permission.Fls) > 0 {
-			p["fls"] = flattenStringSet(permission.Fls)
+		if len(permission.DocumentLevelSecurity) > 0 {
+			p["document_level_security"] = permission.DocumentLevelSecurity
+		}
+		if len(permission.FieldLevelSecurity) > 0 {
+			p["fields_level_security"] = flattenStringSet(permission.FieldLevelSecurity)
```
Comment (phillbaker) [inline es/util.go:326]: Misspelling, should be `field_level_security`

**Comment by JoupainMD** [comment]
Hello, do you have any update about it @phillbaker ?

**Comment by phillbaker** [comment]
Sorry for the delay in getting this merged!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 ### Added
--
+- [open distro role] Add support for OpenDistro document-level-security in role.
 
 ## [1.5.1] - 2020-12-23
 ### Changed
 - Gracefully handle the case where `elasticsearch_index_template` objects exist in the terraform state but not in the ES domain (e.g. because they were manually deleted.)
 - Create index `aliases` and `mappings` even if no settings are set.
 - Bump aws client to v1.35.33.
+<<<<<<< HEAD
```
Comment (phillbaker) [inline CHANGELOG.md:14]: Should remove this.

**Pull request merged**